RUFAS/e2e_test_results_comparer.py

```diff
@@ -91,6 +93,95 @@ def _get_test_result_paths() -> list[ResultPathType]:
         test_result_paths: list[ResultPathType] = []
         for path_set in result_paths:
             test_result_paths.append(
-                ResultPathType(path_set["domain"], path_set["expected_results_path"], path_set["actual_results_path"])
+                ResultPathType(
+                    path_set["domain"],
+                    path_set["expected_results_path"],
+                    path_set["actual_results_path"],
+                    path_set["tolerance"],
+                )
             )
         return test_result_paths
+
+    @staticmethod
+    def is_significant(changes: dict[str, float | str], tolerance: float) -> bool:
+        """
+        Determines if a numerical change is significant based if the change between the
+        "old_value" and "new_value" exceeds the specified tolerance.
+
+        Parameters
+        ----------
+        changes : dict[str, float | str]
+            A dictionary representing changes with "old_value" and "new_value".
+        tolerance : float
+            The threshold for considering a difference as significant.
+
+        Returns
+        -------
+        bool
+            True if the change is both numerical and significant, False otherwise.
+
+        Notes
+        -----
+        The comparison is based on the absolute difference between the "old_value" and "new_value",
+        relative to the "old_value". If the "old_value" is zero, a fallback reference value of 1 is used
+        to ensure the tolerance comparison remains meaningful.
+        """
+        if isinstance(changes, dict) and "old_value" in changes and "new_value" in changes:
+            old_value = changes["old_value"]
+            new_value = changes["new_value"]
+            if isinstance(old_value, (int, float)) and isinstance(new_value, (int, float)):
+                reference = abs(old_value) if abs(old_value) > 0 else 1
+                difference = abs(new_value - old_value)
+                return difference > tolerance * reference
```

Comment:
```suggestion
                return difference > tolerance * GeneralConstants.PERCENT_TO_FRACTION * reference
```
What would you think of adding this, so that a 1% tolerance would be expressed as 1 instead of 0.01? I think it would make the tolerance easier to understand and set for someone who is just reading the description.

Reply:
I'm not sure I understand. If we used this conversion factor we'd also need to do a comparable adjustment on the other side of the equation otherwise we're throwing off the comparison.

Reply:
I think the conversion would be done by the user when specifying the tolerance. So if they had 0.01 as the tolerance before, they would rewrite the tolerance to be 1.0 and the conversion factor would turn it back into 0.01

Reply:
Ah, gotcha. I guess I could see it either way. Do you feel strongly it should be a %? I will implement it. Seems like more frequently tolerance is referred to as a % in general.